src/query_contract_ui/eigenlayer.c

```diff
@@ -0,0 +1,227 @@
+ *
+ */
+bool deposit_into_strategy_ui(ethQueryContractUI_t *msg, context_t *context) {
+    bool ret = false;
+    lr_deposit_t *params = &context->param_data.lr_deposit;
+
+    switch (msg->screenIndex) {
+        case 0:
+            strlcpy(msg->title, "Liquid Restaking", msg->titleLength);
+            strlcpy(msg->msg, "Deposit In Strategy", msg->msgLength);
+            ret = true;
+            break;
+        case 1:
+            strlcpy(msg->title, "Strategy", msg->titleLength);
+            if (params->strategy_to_display == UNKNOW_LR_STRATEGY ||
+                params->strategy_to_display >= LR_STRATEGIES_COUNT) {
+                strlcpy(msg->msg, "UNKNOWN", msg->msgLength);
+            } else {
+                strlcpy(msg->msg, lr_tickers[params->strategy_to_display], MAX_TICKER_LEN);
+            }
+            ret = true;
+            break;
+        case 2:
+            strlcpy(msg->title, "Amount", msg->titleLength);
+            amountToString(params->erc20_amount_to_display,
+                           sizeof(params->erc20_amount_to_display),
+                           ERC20_DECIMALS,
+                           params->erc20_to_display == UNKNOW_LR_STRATEGY ||
+                                   params->erc20_to_display >= LR_STRATEGIES_COUNT
+                               ? "UNKNOWN"
+                               : lr_tickers[params->erc20_to_display],
+                           msg->msg,
+                           msg->msgLength);
+            ret = true;
+            break;
+        default:
+            PRINTF("Received an invalid screenIndex\n");
+            break;
+    }
+    return ret;
+}
+
+/**
+ * @brief UI for queueing withdrawals
+ *
+ * @param msg: message containing the parameter
+ * @param context: context with provide_parameter data
+ *
+ */
+bool queue_withdrawals_ui(ethQueryContractUI_t *msg, context_t *context) {
+    bool ret = false;
+    lr_queue_withdrawals_t *params = &context->param_data.lr_queue_withdrawals;
+
+    switch (msg->screenIndex) {
+        case 0:
+            strlcpy(msg->title, "EigenLayer", msg->titleLength);
+            strlcpy(msg->msg, "Queue Withdrawals", msg->msgLength);
+            ret = true;
+            break;
+
+        case 1:
+            strlcpy(msg->title, "Withdrawer", msg->titleLength);
+            strlcpy(msg->msg, params->withdrawer, msg->msgLength);
+            ret = true;
+            break;
+
+        default: {
+            {
+                // removing the first screen to current screen index
+                // to get the index of the withdrawal
+                uint8_t withdrawal_index = msg->screenIndex - 2;
+
+                if (withdrawal_index < params->strategies_count) {
+                    strlcpy(msg->title, "Strategy", msg->titleLength);
+                    uint8_t strategy = params->strategies[withdrawal_index];
+
+                    if (strategy == UNKNOW_LR_STRATEGY || strategy - 1 >= LR_STRATEGIES_COUNT) {
```

Comment:
it's strange that here we need to `- 1` but in the prior function it was not required